string rel_name
string[] obj_names
bool value